src/Microsoft.SqlTools.ServiceLayer/Cms/CmsService.cs

```diff
@@ -0,0 +1,434 @@
+using Microsoft.SqlTools.ServiceLayer.Cms.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.Connection.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Hosting;
+using Microsoft.SqlTools.Utility;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.SqlTools.ServiceLayer.Cms
+{
+    /// <summary>
+    /// Main class for DacFx service
+    /// </summary>
+    public class CmsService
+    {
+        private static object lockObject = new object();
+        private static ConnectionService connectionService = null;
+        private static RegisteredServersStore registerdServerStore = null;
+        private static readonly Lazy<CmsService> instance = new Lazy<CmsService>(() => new CmsService());
+
+        /// <summary>
+        /// Gets the singleton instance object
+        /// </summary>
+        public static CmsService Instance
+        {
+            get { return instance.Value; }
+        }
+
+        #region Public methods
+
+        /// <summary>
+        /// Initializes the service instance
+        /// </summary>
+        /// <param name="serviceHost"></param>
+        public void InitializeService(ServiceHost serviceHost)
+        {
+            serviceHost.SetRequestHandler(CreateCentralManagementServerRequest.Type, this.HandleCreateCentralManagementServerRequest);
+            serviceHost.SetRequestHandler(ListRegisteredServerRequest.Type, this.HandleListRegisteredServersRequest);
+            serviceHost.SetRequestHandler(AddRegisteredServerRequest.Type, this.HandleAddRegisteredServerRequest);
+            serviceHost.SetRequestHandler(RemoveRegisteredServerRequest.Type, this.HandleRemoveRegisteredServerRequest);
+            serviceHost.SetRequestHandler(AddServerGroupRequest.Type, this.HandleAddServerGroupRequest);
+            serviceHost.SetRequestHandler(RemoveServerGroupRequest.Type, this.HandleRemoveServerGroupRequest);
+        }
+
+        public async Task HandleCreateCentralManagementServerRequest(CreateCentralManagementServerParams createCmsParams, RequestContext<ListRegisteredServersResult> requestContext)
+        {
+            Logger.Write(TraceEventType.Verbose, "HandleCreateCentralManagementServerRequest");
+            try
+            {
+                CmsTask = Task.Run(async () =>
+                {
+                    try
+                    {
+                        //Validate params and connect
+                        ServerConnection conn = await ValidateAndCreateConnection(createCmsParams.ConnectParams);
+
+                        // Get Current Reg Servers on CMS
+                        RegisteredServersStore store = new RegisteredServersStore(conn);
+                        ServerGroup parentGroup = store.DatabaseEngineServerGroup; //TODO Do we need other types like Integrated Service and Analysis Service                    
```

Comment:
we're not supporting SSIS or AS in this codebase, so probably don't need the `TODO`.  If we want that in the future there will need to be large scale refactoring in a bunch of places.